yt/visualization/plot_modifications.py

```diff
@@ -105,7 +105,7 @@ def project_coords(self, plot, coord):
         code_length units.  Projected data units are 2D versions of the
         simulation data units relative to the axes of the final plot.
         """
-        if len(coord) == 3:
+        if coord.shape[0] == 3:
```

Comment:
Is `coord` always going to be an array?  Is it necessary to do something weird like: `if getattr(coord, 'shape', [len(coord)])[0]` to make it work?

Reply:
Isn't the version with `len` a bit more generic than the one with shape, no? Both should return the same value for arrays, but the `len` will work with lists, tuples, ...

Reply:
Just looking at a few places this function is used, it looks like it is exposed to user-supplied data (for example, in the `LinePlotCallback`, that was just the first place I looked, there may be others).

We should probably be making sure all user-supplied data are converted to numpy arrays, in which case the change you've made here would be fine, but that's a much bigger job than just this PR (see https://github.com/yt-project/yt/issues/1839 for more detail about this). If `len` is sufficient, I'd just leave this as-is. If for some reason this PR needs that to change then Matt's suggestion would probably also work.

Reply:
The only place in all of yt that `project_coords()` is called is in `sanitize_coord_system()` and I explicitly changed the first line in that function to assure that coord is a numpy array (see below), so there is never any danger that project_coords gets a non-numpy array object.  

To answer @ngoldbaum 's suggestions, `sanitize_coord_system()` is the first function called in all of the callbacks, and its first line is to convert the coordinates to a numpy array.  So it basically covers all of the problem cases.

When I originally wrote this function, `project_coords()`, I should've made it private, but I didn't, and it seemed reckless to make it private now, but maybe that's appropriate?

Reply:
OK, I went and made those internal methods private.